docs/modules/ROOT/pages/reference/extensions/google-pubsub.adoc

```diff
@@ -32,3 +32,12 @@ Please refer to the above link for usage and configuration details.
 ----
 
 Check the xref:user-guide/index.adoc[User guide] for more information about writing Camel Quarkus applications.
+
+== Camel Quarkus limitations
+
+By default, the Camel PubSub component uses JDK object serialization via `ObjectOutputStream` whenever the message body is anything other than `String` or `byte[]`.
+
+Since such serialization is not yet supported by GraalVM, this extension provides a custom Jackson based serializer to serialize complex message payloads as JSON.
+
+If your payload contains binary data, then you will need to handle that by creating a custom Jackson Serializer / Deserializer.
```

Comment:
I guess the readers could find this link to Quarkus Jackson guide helpful https://quarkus.io/guides/writing-extensions#customizing-jackson where they explain how to write serializers.